server/src/main/java/org/opensearch/common/cache/store/OpenSearchOnHeapCache.java

```diff
@@ -167,6 +185,7 @@ public <K, V> ICache<K, V> create(CacheConfig<K, V> config, CacheType cacheType,
                 .setMaximumWeightInBytes(((ByteSizeValue) settingList.get(MAXIMUM_SIZE_IN_BYTES_KEY).get(settings)).getBytes())
                 .setExpireAfterAccess(((TimeValue) settingList.get(EXPIRE_AFTER_ACCESS_KEY).get(settings)))
                 .setWeigher(config.getWeigher())
+                .setSettings(config.getSettings())
```

Comment:
Why not set the boolean here instead of passing settings object? Looks consistent with the maximumWeight and expireAfterAccess settings. Or we can use the single settings object for reading all 3 within the constructor itself